torch_np/tests/numpy_tests/core/test_multiarray.py

```diff
@@ -2743,29 +2546,7 @@ def test_dot_out_mem_overlap(self):
             assert_raises(ValueError, np.dot, a, b, out=b[::2])
             assert_raises(ValueError, np.dot, a, b, out=b.T)
 
-    def test_dot_matmul_out(self):
-        # gh-9641
-        class Sub(np.ndarray):
-            pass
-        a = np.ones((2, 2)).view(Sub)
-        b = np.ones((2, 2)).view(Sub)
-        out = np.ones((2, 2))
-
-        # make sure out can be any ndarray (not only subclass of inputs)
-        np.dot(a, b, out=out)
-        np.matmul(a, b, out=out)
-
-    def test_dot_matmul_inner_array_casting_fails(self):
-
-        class A:
-            def __array__(self, *args, **kwargs):
-                raise NotImplementedError
-
-        # Don't override the error from calling __array__()
-        assert_raises(NotImplementedError, np.dot, A(), A())
-        assert_raises(NotImplementedError, np.matmul, A(), A())
-        assert_raises(NotImplementedError, np.inner, A(), A())
-
+    @pytest.mark.xfail(reason="TODO [::-1]")
```

Comment:
The issue is this line (ln 2777): `c = np.matmul(a, a, out=a[::-1, ...])`. The intent here is to test matmul with overlapping memory, so I actually wonder if pytorch guarantees this, esp on gpu? If not, it's a skip. If yes, it's a low-prio xfail I think.

Reply:
Note that `[::-1]`'s memory does not overlap.
Now, we can't have `flip` as a view, so `[::-1]` will never work. If what you want is to test whether writing to the `out=` with a weird input that has the right size, that should always work. Even more, I believe the `out=` behaviour should be implemented generically via a copy, as we sort of do now, so that should work for any tensor with non-overlapping memory. I do not know what would be the semantics of `out=` for a tensor with overlapping memory. In PyTorch, it errors out.